### PR TITLE
feat(Guild - Sticker - Thread - Channel - Role) Incluses data 'name' or 'reason' in object

### PR DESCRIPTION
**Pull request by thebigwolfy**
Hello, i am french. Sorry for language ! Thanks you, so, i inclused in function "create" and "createWebhook" the data "name". I am me logic thinking and others change

### CODE REVIEW

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 1]
```diff
@@ -392,7 +394,7 @@ class GuildChannelManager extends CachedManager {
    *   .then(guild => console.log(`Updated channel positions for ${guild}`))
    *   .catch(console.error);
    */
-  async setPositions(channelPositions) {
+  async setPositions(channelPositions = []) {
```
Comment (almeidx) [inline packages/discord.js/src/managers/GuildChannelManager.js:397]: Why was this added?

**Review by xHyroM — CHANGES_REQUESTED** [review]
You need update typings

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 2]
```diff
@@ -175,19 +178,19 @@ class GuildChannelManager extends CachedManager {
...
+   * guild.channels.createWebhook('222197033908436994', {
+   *   name: 'Hello!',
    *   avatar: 'https://i.imgur.com/mI8XcpG.jpg',
    *   reason: 'Needed a cool new Webhook'
    * })
    *   .then(console.log)
    *   .catch(console.error)
    */
-  async createWebhook(channel, name, { avatar, reason } = {}) {
+  async createWebhook(channel, { name, avatar, reason } = {}) {
```
[the remark this answers is not in the document]
Comment (thebigwolfy) [reply packages/discord.js/src/managers/GuildChannelManager.js:193]: Why ?

[the remark this answers is not in the document]
Comment (imranbarbhuiya) [reply packages/discord.js/src/managers/GuildChannelManager.js:193]: > Why ?

name is required

**Comment by thebigwolfy** [comment]
Sorry about all these changes, I got the file wrong ^^'

File: packages/discord.js/src/managers/RoleManager.js [hunk 3]
```diff
@@ -135,8 +135,17 @@ class RoleManager extends CachedManager {
...
+    name,
+    color,
+    hoist,
+    permissions,
+    position,
+    mentionable,
+    reason,
+    icon,
+    unicodeEmoji
+  }) {
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/managers/RoleManager.js:148]:
```suggestion
  } = {}) {
```

All the options are optional

File: packages/discord.js/src/managers/RoleManager.js [hunk 4]
```diff
@@ -118,7 +118,7 @@ class RoleManager extends CachedManager {
   /**
    * Creates a new role in the guild with given information.
    * <warn>The position will silently reset to 1 if an invalid one is provided, or none.</warn>
-   * @param {CreateRoleOptions} [options] Options for creating the new role
+   * @param {CreateRoleOptions} options Options for creating the new role
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/managers/RoleManager.js:121]:
```suggestion
   * @param {CreateRoleOptions} [options] Options for creating the new role
```

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/managers/RoleManager.js [hunk 5]
```diff
@@ -169,20 +178,22 @@ class RoleManager extends CachedManager {
    * Edits a role of the guild.
    * @param {RoleResolvable} role The role to edit
    * @param {RoleData} data The new data for the role
-   * @param {string} [reason] Reason for editing this role
    * @returns {Promise<Role>}
    * @example
    * // Edit a role
-   * guild.roles.edit('222079219327434752', { name: 'buddies' })
-   *   .then(updated => console.log(`Edited role name to ${updated.name}`))
+   * guild.roles.edit('222079219327434752', {
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/managers/RoleManager.js:184]: imo this change isn't needed as reason isn't a role data so it shouldn't belong to `roleData` or u need to change the typedef name to `roleEditData` but I won't suggest it. Also you need to edit the typedef also
https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/src/structures/Role.js#L193

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 6]
```diff
@@ -105,17 +105,20 @@ class GuildChannelManager extends CachedManager {
 
   /**
    * Creates a new channel in the guild.
-   * @param {string} name The name of the new channel
-   * @param {GuildChannelCreateOptions} [options={}] Options for creating the new channel
+   * @param {GuildChannelCreateOptions} options Options for creating the new channel
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/managers/GuildChannelManager.js:108]: you need to update the typedef
https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/src/structures/CategoryChannel.js#L30

**Comment by thebigwolfy** [comment]
> you need to update the typedef

What do I need to change?

**Comment by xHyroM** [comment]
> > you need to update the typedef
> 
> What do I need to change?

For GuildChannelManager: https://github.com/discordjs/discord.js/blob/main/packages/discord.js/typings/index.d.ts#L2769-L2784

For RoleManager: https://github.com/discordjs/discord.js/blob/main/packages/discord.js/typings/index.d.ts#L3008

**Comment by thebigwolfy** [comment]
> > > you need to update the typedef
> > 
> > 
> > What do I need to change?
> 
> For GuildChannelManager:
> 
> https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/typings/index.d.ts#L2769-L2784
> 
> For RoleManager:
> 
> https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/typings/index.d.ts#L3008

I am change, your look commit

**Comment by imranbarbhuiya** [comment]
Not only typings I mean typedef 

https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/src/structures/CategoryChannel.js#L30
https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/src/structures/Role.js#L193

**Comment by imranbarbhuiya** [comment]
Also run prettier. You've changed the formatting of typings file

File: packages/discord.js/typings/index.d.ts [hunk 7]
```diff
@@ -2836,16 +2836,14 @@ export class GuildChannelManager extends CachedManager<Snowflake, GuildBasedChan
   ): Promise<MappedGuildChannelTypes[T]>;
   /** @deprecated See [Self-serve Game Selling Deprecation](https://support-dev.discord.com/hc/en-us/articles/4414590563479) for more information */
   public create(
```
Comment (xHyroM) [inline packages/discord.js/typings/index.d.ts:2838]: You need update `GuildChannelCreateOptions` too (add name property).
Same for `ChannelData` and `RoleData` but you need add reason

**Comment by thebigwolfy** [comment]
> Not only typings I mean typedef
> 
> https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/src/structures/CategoryChannel.js#L30
> 
> 
> https://github.com/discordjs/discord.js/blob/861f0e2134662ab64a11d313130aff58b413d495/packages/discord.js/src/structures/Role.js#L193

Fix

File: packages/discord.js/src/structures/BaseGuildVoiceChannel.js [hunk 8]
```diff
@@ -91,8 +91,8 @@ class BaseGuildVoiceChannel extends GuildChannel {
    * // Remove a fixed region for this channel - let Discord decide automatically
    * channel.setRTCRegion(null);
    */
-  setRTCRegion(region) {
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/structures/BaseGuildVoiceChannel.js:94]: 🤔

File: packages/discord.js/src/structures/CategoryChannel.js [hunk 9]
```diff
@@ -52,7 +52,8 @@ class CategoryChannel extends GuildChannel {
    * @returns {Promise<GuildChannel>}
    */
   createChannel(name, options) {
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/CategoryChannel.js:54]: Shouldn't this also move name in the options object?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Comment by thebigwolfy** [comment]
I think I've changed most things, do you think there are still some?

File: packages/discord.js/src/structures/BaseGuildVoiceChannel.js [hunk 10]
```diff
@@ -91,8 +91,8 @@ class BaseGuildVoiceChannel extends GuildChannel {
...
-    return this.edit({ rtcRegion: region });
+  setRTCRegion(rtcRegion) {
```
Comment (xHyroM) [inline packages/discord.js/src/structures/BaseGuildVoiceChannel.js:94]: You must also rename the parameter in `index.d.ts`

https://github.com/discordjs/discord.js/blob/d32db8833e1058fb36f2e83af79d5353a9f2f693/packages/discord.js/typings/index.d.ts#L434

**Review by xHyroM — CHANGES_REQUESTED** [review]

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/managers/CategoryChannelChildManager.js [hunk 11]
```diff
@@ -54,12 +54,11 @@ class CategoryChannelChildManager extends DataManager {
   /**
    * Creates a new channel within this category.
    * <info>You cannot create a channel of type {@link ChannelType.GuildCategory} inside a CategoryChannel.</info>
-   * @param {string} name The name of the new channel
    * @param {CategoryCreateChannelOptions} options Options for creating the new channel
```
Comment (almeidx) [inline packages/discord.js/src/managers/CategoryChannelChildManager.js:57]: The `CategoryCreateChannelOptions` typedef doesn't include `name`

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 12]
```diff
@@ -175,19 +176,19 @@ class GuildChannelManager extends CachedManager {
   /**
    * Creates a webhook for the channel.
    * @param {GuildChannelResolvable} channel The channel to create the webhook for
-   * @param {string} name The name of the webhook
-   * @param {ChannelWebhookCreateOptions} [options] Options for creating the webhook
+   * @param {ChannelWebhookCreateOptions} options Options for creating the webhook
```
Comment (almeidx) [inline packages/discord.js/src/managers/GuildChannelManager.js:179]: The `ChannelWebhookCreateOptions` typedef doesn't include `name`

File: packages/discord.js/src/managers/GuildChannelManager.js [hunk 13]
```diff
@@ -228,21 +229,23 @@ class GuildChannelManager extends CachedManager {
    * Edits the channel.
    * @param {GuildChannelResolvable} channel The channel to edit
    * @param {ChannelData} data The new data for the channel
-   * @param {string} [reason] Reason for editing this channel
```
Comment (almeidx) [inline packages/discord.js/src/managers/GuildChannelManager.js:231]: The `ChannelData` typedef doesn't include `reason`

File: packages/discord.js/src/managers/RoleManager.js [hunk 14]
```diff
@@ -169,20 +178,22 @@ class RoleManager extends CachedManager {
    * Edits a role of the guild.
    * @param {RoleResolvable} role The role to edit
    * @param {RoleData} data The new data for the role
-   * @param {string} [reason] Reason for editing this role
```
Comment (almeidx) [inline packages/discord.js/src/managers/RoleManager.js:172]: The `RoleData` typedef doesn't include `reason`

File: packages/discord.js/src/structures/BaseGuildVoiceChannel.js (hunk 10 above)
Comment (almeidx) [reply packages/discord.js/src/structures/BaseGuildVoiceChannel.js:94]: Need to change the parameter name in the jsdoc above the method

File: packages/discord.js/src/structures/Guild.js [hunk 15]
```diff
@@ -773,17 +773,17 @@ class Guild extends AnonymousGuild {
   /**
    * Updates the guild with new information - e.g. a new name.
    * @param {GuildEditData} data The data to update the guild with
-   * @param {string} [reason] Reason for editing this guild
```
Comment (almeidx) [inline packages/discord.js/src/structures/Guild.js:776]: The `GuildEditData` typedef doesn't include `reason`

File: packages/discord.js/src/structures/Role.js [hunk 16]
```diff
@@ -426,6 +428,7 @@ class Role extends Base {
       permissions: this.permissions.toJSON(),
     };
   }
+
```
Comment (almeidx) [inline packages/discord.js/src/structures/Role.js:431]: Why was this new line added?

**Review by almeidx — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Guild.js [hunk 17]
```diff
@@ -826,7 +826,7 @@ class Guild extends AnonymousGuild {
     }
     if (data.preferredLocale) _data.preferred_locale = data.preferredLocale;
     if ('premiumProgressBarEnabled' in data) _data.premium_progress_bar_enabled = data.premiumProgressBarEnabled;
-    const newData = await this.client.rest.patch(Routes.guild(this.id), { body: _data, reason });
+    const newData = await this.client.rest.patch(Routes.guild(this.id), { body: _data, data.reason });
```
Comment (almeidx) [inline packages/discord.js/src/structures/Guild.js:829]:
```suggestion
    const newData = await this.client.rest.patch(Routes.guild(this.id), { body: _data, reason: data.reason });
```

File: packages/discord.js/typings/index.d.ts [hunk 18]
```diff
@@ -2815,16 +2815,14 @@ export class GuildChannelManager extends CachedManager<Snowflake, GuildBasedChan
...
-    name: string,
     options: GuildChannelCreateOptions & { type: ChannelType.GuildStore },
   ): Promise<StoreChannel>;
   public create(name: string, options?: GuildChannelCreateOptions): Promise<TextChannel>;
```
Comment (almeidx) [inline packages/discord.js/typings/index.d.ts:2820]:
```suggestion
  public create(options: GuildChannelCreateOptions): Promise<TextChannel>;
```

File: packages/discord.js/typings/index.d.ts [hunk 19]
```diff
@@ -2815,16 +2815,14 @@ export class GuildChannelManager extends CachedManager<Snowflake, GuildBasedChan
...
   public createWebhook(
     channel: GuildChannelResolvable,
-    name: string,
     options?: ChannelWebhookCreateOptions,
```
Comment (almeidx) [inline packages/discord.js/typings/index.d.ts:2823]:
```suggestion
    options: ChannelWebhookCreateOptions,
```

File: packages/discord.js/typings/index.d.ts [hunk 20]
```diff
@@ -4794,6 +4795,7 @@ export interface ResolvedOverwriteOptions {
 }
 
 export interface RoleData {
+  reason: string;
```
Comment (almeidx) [inline packages/discord.js/typings/index.d.ts:4798]: The reason should be optional. Also, you need to change the `Role#edit()` typings

```suggestion
  reason?: string;
```

Comment (almeidx) [reply packages/discord.js/typings/index.d.ts:4798]: Besides this:
- the `CreateRoleOptions` interface extends this one just to add the `reason` parameter is now obsolete 
- the `PartialRoleData` interface extends this one but should not have a `reason` parameter

File: packages/discord.js/typings/index.d.ts [hunk 21]
```diff
@@ -2815,16 +2815,14 @@ export class GuildChannelManager extends CachedManager<Snowflake, GuildBasedChan
   ): Promise<MappedGuildChannelTypes[T]>;
   /** @deprecated See [Self-serve Game Selling Deprecation](https://support-dev.discord.com/hc/en-us/articles/4414590563479) for more information */
   public create(
```
Comment (almeidx) [inline packages/discord.js/typings/index.d.ts:2817]: You need to remove the `name` parameter from the overload above this one (I can't comment there)

File: packages/discord.js/src/structures/ThreadChannel.js [hunk 22]
```diff
@@ -360,13 +363,11 @@ class ThreadChannel extends Channel {
    * @example
    * // Set the thread's auto archive time to 1 hour
    * thread.setAutoArchiveDuration(60)
-   *   .then(newThread => {
-   *     console.log(`Thread will now archive after ${newThread.autoArchiveDuration} minutes of inactivity`);
-   *    });
+   *   .then(newThread => console.log(`Thread will now archive after ${newThread.autoArchiveDuration} minutes of inactivity`));
```
Comment (almeidx) [inline packages/discord.js/src/structures/ThreadChannel.js:366]: This line is too long, and the linter will complain about it. That's why it was setup the way it was beforehand

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by kyranet** [comment]
You have formatting errors, please run Prettier.

**Comment by thebigwolfy** [comment]
> You have formatting errors, please run Prettier.

What is Prettier? I don't know and therefore don't use it ^^'

**Comment by kyranet** [comment]
@thebigwolfy please install dependencies, I have no idea what you're using, but Git should run Prettier before making the commit, making committing unformated code impossible, but that's not the case here.

As the CI displays, this PR has several lint errors:

```
discord.js:lint: [warn] src/managers/GuildMemberManager.js
discord.js:lint: [warn] src/managers/GuildScheduledEventManager.js
discord.js:lint: [warn] src/managers/RoleManager.js
discord.js:lint: [warn] src/structures/GuildChannel.js
discord.js:lint: [warn] Code style issues found in the above file(s). Forgot to run Prettier?
```

Prettier is the name of the tool we use to format, it's installed automatically as a development dependency, you run `yarn fmt` to run it if outside of the commit hook, and nothing else.

**Comment by thebigwolfy** [comment]
I don't use Prettier, I can't use it ^^' @kyranet

**Comment by vladfrangu** [comment]
You can use it by installing dependencies then running the script kyra told you (`yarn fmt`)

**Comment by thebigwolfy** [comment]
> You can use it by installing dependencies then running the script kyra told you (`yarn fmt`)

I must be too bad, I can't do it :/

**Review by SpaceEEC — APPROVED** [review]

File: packages/discord.js/src/managers/GuildMemberManager.js [hunk 23]
```diff
@@ -250,10 +255,9 @@ class GuildMemberManager extends CachedManager {
    * <info>The user must be a member of the guild</info>
    * @param {UserResolvable} user The member to edit
    * @param {GuildMemberEditData} data The data to edit the member with
-   * @param {string} [reason] Reason for editing this user
    * @returns {Promise<GuildMember>}
    */
-  async edit(user, data, reason) {
+  async edit(user, data) {
```
Comment (muchnameless) [inline packages/discord.js/src/managers/GuildMemberManager.js:260]: calling edit with `{ nick, reason }` as data would use the wrong route ([again](https://github.com/discordjs/discord.js/pull/7151)), the check in [lines 287-289](https://github.com/discordjs/discord.js/pull/7448/files#diff-601b8954b3afea4b089cd0afd7b1c9679c874acaa1ea7c5a658319b5312e856aR287-R289) needs to be adapted

File: packages/discord.js/src/managers/GuildScheduledEventManager.js [hunk 24]
```diff
@@ -185,25 +183,22 @@ class GuildScheduledEventManager extends CachedManager {
...
+    description,
+    scheduledEndTime,
+    entityMetadata,
+    reason,
+    image,
+  }) {
     const guildScheduledEventId = this.resolveId(guildScheduledEvent);
     if (!guildScheduledEventId) throw new Error('GUILD_SCHEDULED_EVENT_RESOLVE');
 
-    if (typeof options !== 'object') throw new TypeError('INVALID_TYPE', 'options', 'object', true);
```
Comment (AverageHelper) [inline packages/discord.js/src/managers/GuildScheduledEventManager.js:192]: This may be nothing, as I may have missed some important note about discord.js' policy on handling runtime type issues going forward, but I figured I'd bring this up just in case:

Isn't removing this type check (and similar ones in this PR) a functional regression? If someone (who's not using any static type-checking tools) passes a non-object in for `options`, I'd assume the runtime errors would be about the destructured properties being `undefined`, as opposed to the previous easy-to-understand `TypeError` against the parameter as a whole.

**Comment by thebigwolfy** [comment]
> @thebigwolfy Please resolve the linting issues, if you are facing problems tell us

It's done, desolate the expectation ^^' I had trouble merging on my commit...

**Comment by thebigwolfy** [comment]
Fix all bug, doc, ect. Sorry for error in typscript ^^'
I have used #8026 for resolve error ect
